### PR TITLE
Remove site and workflows, update README

### PR DESCRIPTION
**Pull request by jaimergp**
Follow up to #171 and https://github.com/conda-forge/conda-forge.github.io/pull/2090.

Since the status page is now rendered in the main website repo, we can remove this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,27 +1,22 @@
...
   - degraded performance
   - major outage
 
-Then GitHub Actions will update the page with the issue title and the
-text of the issue body.
+The [status page](https://conda-forge.org/status/) will automatically fetch the issues
+and render them as incidents.
 
-When the incident is resolved, close the issue and the status page will
-return to normal.
```
Comment (beckermr) [inline README.md]: Let's keep this unless you think it is too obvious?

Comment (jakirkham) [reply README.md]: Agreed

Comment (jaimergp) [reply README.md]: Oops, yes, this was an accident. Reverted!

**Comment by jakirkham** [comment]
Thank you both! 🙏

Should we drop this too?

https://github.com/conda-forge/status/blob/0c4c038b25c566a3f0b832a8e7f5842e346602fe/.gitignore#L1

**Comment by jaimergp** [comment]
I usually leave the old gitignore entries so folks who had it cloned don't accidentally commit back the files they might have from previous checkouts.

**Comment by jakirkham** [comment]
Interesting ok

Though that seems like a non-issue here, right?

**Pull request merged**